presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -856,10 +856,18 @@ protected Object visitNotExpression(NotExpression node, Object context)
         @Override
         protected Object visitLogicalBinaryExpression(LogicalBinaryExpression node, Object context)
         {
+            LogicalBinaryExpression.Type type = node.getType();
             Object left = process(node.getLeft(), context);
+            if (type == LogicalBinaryExpression.Type.AND && Boolean.FALSE.equals(left)) {
+                return false;
+            }
+            if (type == LogicalBinaryExpression.Type.OR && Boolean.TRUE.equals(left)) {
+                return true;
+            }
+
             Object right = process(node.getRight(), context);
 
-            switch (node.getType()) {
+            switch (type) {
                 case AND: {
                     // if either left or right is false, result is always false regardless of nulls
                     if (Boolean.FALSE.equals(left) || Boolean.TRUE.equals(right)) {
```

Comment:
Comment: "if either left or right is false...",
looks detached from the actual code: "FALSE.equals(left) || TRUE.equals(right)".

(code is actually correct, "smarter" than the comment)

Would you consider removing this comment? (Same for OR case)